test/jdk/javax/swing/JFileChooser/4525475/JFileChooserReadOnlyTest.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2004, 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Component;
+import java.awt.Container;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+import javax.swing.Action;
+import javax.swing.JButton;
+import javax.swing.JFileChooser;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.UIManager.LookAndFeelInfo;
+import javax.swing.UnsupportedLookAndFeelException;
+
+import static javax.swing.UIManager.getInstalledLookAndFeels;
+
+/*
+ * @test
+ * @key headful
+ * @bug 4525475
+ * @summary This testcase tests JDK-4525475 bug fix, checks whether JFileChooser
+ *          allows modification to the file-system by way of the "New Folder"
+ *          button or not, ideally a read-only JFileChooser shouldn't allow it.
+ * @run main JFileChooserReadOnlyTest
+ */
+public class JFileChooserReadOnlyTest {
+
+    private static volatile boolean result = true;
+    private static volatile boolean newFolderFound = false;
+
+    public static void main(String[] args) throws Exception {
+
+        List<String> lafs = Arrays.stream(getInstalledLookAndFeels())
+                                  .map(LookAndFeelInfo::getClassName)
+                                  .collect(Collectors.toList());
+        for (final String laf : lafs) {
+            if (!setLookAndFeel(laf)) {
+                continue;
+            }
+
+            // Test1, Read Only JFileChooser
+            final AtomicBoolean readOnly = new AtomicBoolean(true);
+            SwingUtilities.invokeAndWait(
+                    () -> createAndTestCustomFileChooser(readOnly.get()));
+            System.out.println("Its a Read Only JFileChooser " +
```

Comment:
```suggestion
            System.out.println("It's a read-only JFileChooser " +
```

The apostrophe is missing.